src/iop/highlights.c

```diff
@@ -2034,8 +2076,10 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
                   fminf(piece->pipe->dsc.processed_maximum[1], piece->pipe->dsc.processed_maximum[2]));
     }
     else
-      _process_linear_opposed(piece, ivoid, ovoid, roi_in, roi_out, data);
-
+    {
+      float *tmp = _process_linear_opposed(piece, ivoid, ovoid, roi_in, roi_out, data);
```

Comment:
We have a single call to _process_linear_opposed(), which return a pointer that is not used and freed directly. Can't we avoid return it? Or is that for some further enhancement that you are planning?

Reply:
I still hope to get more sraw images to test, so far results are good but I won't integrate segmentation until proper checks have been done.